deploy-board/deploy_board/webapp/deploy_views.py

```diff
@@ -28,6 +28,36 @@
 DEFAULT_PAGE_SIZE = 30
 DEFAULT_ONGOING_DEPLOY_SIZE = 10
 
+def _get_running_deploys_count(request):
+    deploy_states = ["RUNNING"]  # current running deploys
+    page_size = 1                # only need to return 1 detail, the return structure has the "total" value
+    deployResult = deploys_helper.get_all(request, deployState=deploy_states, pageSize=page_size)
+    return deployResult['total']
+
+def _get_sidecars(request):
+    # returns a list of env id for sidecars which are identified by having a system priority
+    envs = environs_helper.get_all_sidecar_envs(request)
+    env_ids = []
+    for env in envs:
+        env_ids.append(env['id'])
+    return env_ids
+
+def _get_ongoing_sidecar_deploys(request):
+    deploy_summaries = []
+    env_ids = _get_sidecars(request)
+    if env_ids:
+        deploy_states = ["RUNNING", "FAILING"]
+        deployResult = deploys_helper.get_all(request, envId=env_ids, deployState=deploy_states)
```

Comment:
Although it does not matter for sidecar today, Can we not load list of all services or sidecars on on homepage? My suggestion is load on demand (explicitly clicked by user)

Reply:
worried about get_all kind of calls slowing UI especially when it's not paginated.

Reply:
it is a filtered call for limited sidecar envs. I also reduced the refreshed interval from 30sec to 60sec. if it turn out the queries are too expensive (with growing deploys tables), we can: 1. move old deploy history to archived table. 2. move the deploy stats into a separate page instead of home page.  as we are re-inventing the Teletraan UI, will delay any extensive UI changes

Reply:
sure, np.